netplugin/plugin/netplugin.go

```diff
@@ -241,6 +288,7 @@ func (p *NetPlugin) Reinit(cfg Config) {
 	}
 
 	defer func() {
```

Comment:
If the function returns before hitting this line, would we end up still holding the lock?